File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        start = 0
        end = int(sqrt(c))

        while(start <= end):
            value = (start*start) + (end*end)
            if(value == c):
                return True
            elif(value > c):
                end = end-1
            else:
                start = start+1
        
        return False
        

File: 633-sum-of-square-numbers/sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        int end = int(sqrt(c));
        int start = 0;

        while(start <= end) {
            long long value = 1LL * start * start + 1LL * end * end;
            cout << value;
            if(value == c) {
                return true;
            } else if(value > c) {
                end--;
            } else {
                start++;
            }
        }

        return false;
    }
};